File: chomik-delivery/src/main/java/com/chomik/chomikdelivery/service/dto/CreateUserAddressRequest.java
package com.chomik.chomikdelivery.service.dto;

import com.fakecdek.delivery.mock.model.dto.DeliveryAddressDto;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record CreateUserAddressRequest (
        @NotBlank(message = "userId must not be null or blank")
        String userId,
        @NotNull(message = "address must not be null or blank")
        @Valid
        DeliveryAddressDto address)
{}

File: chomik-clients/chomik-delivery-client/src/main/kotlin/com/chomik/delivery/client/dto/CreateUserAddressRequest.kt
package com.chomik.delivery.client.dto

import com.fakecdek.delivery.mock.model.dto.DeliveryAddressDto
import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class CreateUserAddressRequest(
    @NotBlank(message = "userId must not be null or blank")
    val userId: String,
    @NotNull(message = "address must not be null or blank")
    @Valid
    val address: DeliveryAddressDto
)
